doc/examples/segment_extending_clustering_framework.py

```diff
@@ -0,0 +1,233 @@
+"""
+===========================================================
+How to make new `Feature` and new `Metric` for Quickbundles
```

Comment:
Hi @MarcCote, this title is not very attractive. 
I would say something more general. For example,
Enhancing QuickBundles with different metrics and features.
or
Introducing the advanced capabilities of the clustering framework. A use case
with adding new metrics to Quickbundles.